pkg/codegen/docs/package_tree.go

```diff
@@ -0,0 +1,75 @@
+package docs
+
+import (
+	"github.com/pkg/errors"
+)
+
+type entryType string
+
+const (
+	entryTypeModule   entryType = "module"
+	entryTypeResource entryType = "resource"
+	entryTypeFunction entryType = "function"
+)
+
+// PackageTreeItem is a type for representing a package in a
+// navigable tree format starting from the top-level/index/root
+// of a package.
+type PackageTreeItem struct {
+	Name     string            `json:"name"`
+	Type     entryType         `json:"type"`
+	Link     string            `json:"link"`
+	Children []PackageTreeItem `json:"children,omitempty"`
+}
+
+func generatePackageTree(rootMod modContext) ([]PackageTreeItem, error) {
```

Comment:
I could also make this a receiver function of the `modContext` type if you think that reads better. I started going down this route and didn't think it was bad enough to have to change to that.